backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/blobs/CassandraBlobMapperTest.java

```diff
@@ -0,0 +1,92 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra.blobs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.datastax.driver.core.utils.UUIDs;
+import com.google.common.base.Charsets;
+import com.google.common.base.Strings;
+
+public class CassandraBlobMapperTest {
+
+    private CassandraCluster cassandra;
+    private CassandraBlobMapper testee;
+
+    @Before
+    public void setUp() {
+        cassandra = CassandraCluster.create(new CassandraBlobModule());
+        cassandra.ensureAllTables();
+
+        testee = new CassandraBlobMapper(cassandra.getConf());
+    }
+
+    @After
+    public void tearDown() {
+        cassandra.clearAllTables();
+        cassandra.close();
+    }
+
+    @Test
+    public void test() throws IOException {
+        byte[] bytes = IOUtils.toByteArray(testee.read(UUIDs.timeBased()));
```

Comment:
So if the blob does not exist we return a empty byte instead of Optional.empty() ? Why ?